File: RpnCalculator-Scala/src/rpn/scala/operators/ClearOperator.scala
package rpn.scala.operators

class ClearOperator extends FullStackOperator {
  val C = "C"
  val CLEAR = "CLEAR"

  def continueOperator(value: Int): Int = { 0; }

  def handlesOperatorCharacter(operand: String): Boolean = {
    operand.toUpperCase() == C || operand.toUpperCase() == CLEAR
  }

}

File: src/rpn/operator/ClearOperator.java
package rpn.operator;

import rpn.RpnStack;

public class ClearOperator implements Operator {

	private static final String C = "C";
	private static final String CLEAR = "CLEAR";

	@Override
	public int doOperation(RpnStack numbers) {
		while (!numbers.isEmpty())
			numbers.pop();
		return 0;
	}

	@Override
	public boolean handlesOperatorCharacter(String operand) {
		return C.contentEquals(operand.toUpperCase()) || CLEAR.contentEquals(operand.toUpperCase());

	}

}
